src/adapters/mbox.rs

```diff
@@ -0,0 +1,238 @@
+use super::*;
+
+use anyhow::Result;
+use async_stream::stream;
+use lazy_static::lazy_static;
+use mime2ext::mime2ext;
+use regex::bytes::Regex;
+use tokio::io::AsyncReadExt;
+
+use std::{collections::VecDeque, io::Cursor};
+
+static EXTENSIONS: &[&str] = &["mbox", "mbx", "eml"];
+static MIME_TYPES: &[&str] = &["application/mbox", "message/rfc822"];
+lazy_static! {
+    static ref METADATA: AdapterMeta = AdapterMeta {
+        name: "mail".to_owned(),
+        version: 1,
+        description:
+            "Reads mailbox/mail files and runs extractors on the contents and attachments."
+                .to_owned(),
+        recurses: true,
+        fast_matchers: EXTENSIONS
+            .iter()
+            .map(|s| FastFileMatcher::FileExtension(s.to_string()))
+            .collect(),
+        slow_matchers: Some(
+            MIME_TYPES
+                .iter()
+                .map(|s| FileMatcher::MimeType(s.to_string()))
+                .collect()
+        ),
+        disabled_by_default: true,
+        keep_fast_matchers_if_accurate: true
+    };
+    static ref FROM_REGEX: Regex = Regex::new("\r?\nFrom [^\n]+\n").unwrap();
+}
+#[derive(Default)]
+pub struct MboxAdapter;
+
+impl MboxAdapter {
+    pub fn new() -> MboxAdapter {
+        MboxAdapter
+    }
+}
+impl GetMetadata for MboxAdapter {
+    fn metadata(&self) -> &AdapterMeta {
+        &METADATA
+    }
+}
+
+#[async_trait]
+impl FileAdapter for MboxAdapter {
+    async fn adapt(
+        &self,
+        ai: AdaptInfo,
+        _detection_reason: &FileMatcher,
+    ) -> Result<AdaptedFilesIterBox> {
+        let AdaptInfo {
+            filepath_hint,
+            mut inp,
+            line_prefix,
+            archive_recursion_depth,
+            config,
+            postprocess,
+            ..
+        } = ai;
+
+        let mut content = Vec::new();
+        let s = stream! {
+            inp.read_to_end(&mut content).await?;
```

Comment:
reading the whole file into RAM is not good because it may be many gigabytes in size (e.g. firefox mailbox files?)